numpy/core/_dtype.py

```diff
@@ -24,11 +24,11 @@
 def _kind_name(dtype):
     try:
         return _kind_to_stem[dtype.kind]
-    except KeyError:
+    except KeyError as e:
         raise RuntimeError(
             "internal dtype error, unknown kind {!r}"
             .format(dtype.kind)
-        )
+        ) from e
```

Comment:
I'd updated https://github.com/numpy/numpy/issues/15986 to be a little clearer - this is a case where `from None` would be better.